### PR TITLE
README improvement

### LINKED ISSUE #10: Fix README according to the latest main branch
Before v0.3.0 release README should be fixed.

### LINKED ISSUE #11: Make develop branch for pre-release commits

### PR DESCRIPTION
**Pull request by minamorl**
Closes #10 and closes #11

### CODE REVIEW

**Comment by minamorl** [comment]
@PhilippMDoerner Can you check if my understanding is correct?

**Comment by PhilippMDoerner** [comment]
Almost, the subscription examples should also contain a `doWork()` at the end or store the subscription and call doWork on it somewhere else. That is part of the async-ness of the implementation and likely should be mentioned explicitly.

There's essentially 2 ways to use the procs that interact with observables: Do the work now (Use "blocking" versions of procs) or do the work later (get the Future for that work from the "async" version and waitFor it later/call `doWork` on a subscription).

This applies to the following procs:

- **Subscribe**: Returns a Subscription that contains the Future representing the async-work that gets prepared **upon subscribing**. Therefore this is only necessary for cold observables (and Behavior/Replaysubjects later). To do that async work you **must** call `subscription.doWork()` somewhere. doWork can also be called directly after subscribe and also returns the subscription for convenience. Check out the test-suite for examples
- **Subject.complete**:  The complete proc is async by default and returns a future of all the remaining async-work of all the complete procs of its observers. You must waitFor that Future somewhere *or* use `completeBlock`
- **Subject.next**: The next proc is async by default and returns a future of all the remaining async-work of all the next/error procs of its observers. You must waitFor that Future somewhere **or** use `nextBlock`

It might make sense to point out that these can be used in these 2 different ways (sync and async) and potentially explain what work each of the futures represents..

File: README.md [hunk 1]
```diff
@@ -1,61 +1,111 @@
...
-myObservable.subscribe(proc(value: int) =
+let subscription = myObservable.subscribe(
+  proc(value: int) =
     echo "Received value: ", value
-)
+).doWork()
 ```
 
-## Example
+Note that calling doWork() is only necessary for cold observables and behavior/replay subjects.
```
Comment (PhilippMDoerner) [inline README.md]: Those don't exist yet, no reason to copy paste that part of my response ;-)

They will eventually, mind you, just not for now, as the current implementation tries to focus on the basics and having a solid fundament for the things yet to come.

**Review by PhilippMDoerner — APPROVED** [review]
LGTM

**Pull request merged**